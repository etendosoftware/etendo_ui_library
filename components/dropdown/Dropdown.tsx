import React, {useState, useRef, useEffect} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  Image,
  ScrollView,
  Platform,
} from 'react-native';

import DropdownItem from './DropdownItem';
import {DropdownStyleVariant} from './Dropdown.styles';
import {DropdownProps, Info} from './Dropdown.types';
import {arrowPickerVerticalIcon} from '../../assets/images/icons';

const Dropdown = ({
  data,
  onChangeSelected,
  text,
  typeSizeText,
}: DropdownProps) => {
  const [showOptions, setShowOptions] = useState(false);
  const [toggleItem, setToggleItem] = useState<any>(undefined);
  const [chooseOption, setChooseOption] = useState(text);

  const PLATFORM_IS_WEB = Platform.OS === 'web';
  const catMenu = useRef(null);

  const handleShowOptions = () => {
    setShowOptions(!open);
  };

  useEffect(() => {
    setChooseOption(text);
  }, [text]);

  const closeOpenMenus = (e: any) => {
    if (catMenu.current && showOptions && !catMenu.current.contains(e.target)) {
      setShowOptions(false);
    }
  };

  const showOptionsIcon = (isWeb: boolean) => {
    return isWeb ? (
      <img
        src={arrowPickerVerticalIcon}
        style={{
          width: 10,
          height: 7,
          position: 'absolute',
          right: 15,
          rotate: '180deg',
        }}
      />
    ) : (
      <Image
        source={arrowPickerVerticalIcon}
        style={{
          width: 11,
          height: 9,
          resizeMode: 'stretch',
          position: 'absolute',
          right: 15,
          transform: [{rotate: '180deg'}],
        }}
      />
    );
  };

  const notShowOptionsIcon = (isWeb: boolean) => {
    return isWeb ? (
      <img
        src={arrowPickerVerticalIcon}
        style={{width: 10, height: 7, position: 'absolute', right: 15}}
      />
    ) : (
      <Image
        source={arrowPickerVerticalIcon}
        style={{
          width: 11,
          height: 9,
          resizeMode: 'stretch',
          position: 'absolute',
          right: 15,
        }}
      />
    );
  };

  document.addEventListener('mousedown', closeOpenMenus);

  return (
    <View style={DropdownStyleVariant.primary.container} ref={catMenu}>
      <TouchableOpacity
        style={DropdownStyleVariant.primary.dropDownButton}
        activeOpacity={1}
        onPress={(e: any) => {
          setShowOptions(!showOptions);
        }}
      >
        <Text style={DropdownStyleVariant.primary.dropDownButtonText}>
          {chooseOption}
        </Text>
        {showOptions
          ? showOptionsIcon(PLATFORM_IS_WEB)
          : notShowOptionsIcon(PLATFORM_IS_WEB)}
      </TouchableOpacity>

      {showOptions && (
        <ScrollView style={DropdownStyleVariant.primary.containerOptions}>
          {data.map((item: Info, index: number) => {
            return (
              <DropdownItem
                item={item}
                index={index}
                onPress={() => {
                  /* Allows to detect the selected item */
                  setToggleItem(index);
                  onChangeSelected(item.name);
                  setChooseOption(item.name);
                  setToggleItem(index);
                  handleShowOptions();
                }}
                key={item.key}
                toggleItem={toggleItem}
                sizeText={typeSizeText}
              />
            );
          })}
        </ScrollView>
      )}
    </View>
  );
};

export default Dropdown;
