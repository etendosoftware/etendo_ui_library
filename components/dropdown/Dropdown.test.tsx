import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

const dataset = [
  {name: 'User', route: '', key: 'user'},
  {name: 'Vehicles', route: '', key: 'vehicles'},
  {name: 'Skills', route: '', key: 'skills'},
  {name: 'Mobile app', route: '', key: 'movilApp'},
  {name: 'Staff', route: '', key: 'staff'},
  {name: 'Routing options', route: '', key: 'routing options'},
  {name: 'Map', route: '', key: 'map'},
];

const onCheckSelectedMock = () => dataset[0];

describe('Running Test for Tab', () => {
  it('Check Tab Disabled', () => {
    const tree = renderer
      .create(
        <Dropdown
          data={dataset}
          onChangeSelected={onCheckSelectedMock}
          text="Massive load"
          style="primary"
          typeSizeText="medium"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
