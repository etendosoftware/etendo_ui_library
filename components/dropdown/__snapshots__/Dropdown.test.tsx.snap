// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Pagination Check Pagination Disabled 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#202452",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 34,
        "opacity": 1,
        "paddingLeft": 10,
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#202452",
          "fontSize": 13,
          "fontWeight": "500",
        }
      }
    >
      Large
    </Text>
    <Image
      source={
        Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAFlCAYAAACdqVCOAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAABmLSURBVHgB7d09bFxnuh/w54y0QNJcDOkmqTKLlSGyWrlzt1RnVyEXuYBTWSpdrdy5W7FLt1KXzlaVABew5lb3ppLUOZV5K40hKznbyYBBHXUKRM7JOYeSSFMjipzP8/H7AQJHFGHLlOD3P+/7f96TBBMNBtv9+F0MInrXIo9+8eOPkRQfI+9HkgyqLyo/H9UPAOooibT6mOevPyZp8bm/V5/Px2kcxF6aDrPooCQ4Wuwvx7VqsY/4U/FdKRf9QQDQBUUAyPeK/+//W/HxYRxWoSCNlutsABhc2d4qkuB28R34U/FtuBYA8Ea1Y5A8jBg/KgLBwzYGgs4EgLfv8vPel8V/9XbYugfg3IqdgTy/16Yw0PoAcPxOP/kyLPoAzOwoDKRPh99Fg7UyABy92+/dKP6QvrS9D8BCvDkmOBzvNnFXoFUB4PU2/18ierfCu30AliWP76I3vps+Ge5FQ7QiAFj4AaiH4njgIL/ZhB2BRgcACz8AtVTuCNT8aKCxAWDw8Z9vFN/gv4WFH4DaGt+Jg7hbxyDQuAAw2NgexEHybfFb3woAqLuqLJjv1m1qoFEBYHBl+6+2+wFopJodCzQiAHjXD0Ar1Gg34FLUXPGu/1aMe/+jWPw3AgCaLEnKB8pt99ev9vv/sPG/s2z0MlaktjsArxv+b7b8AaBdyt2Aw/z6qo4EarkDUG35R/IvxeK/HQDQRtVuQBS7ARt/z56PRrFktQsAg4+3ywf2/IstfwBa7+hI4IviSCCy/dGjWKJaBYDqwT1RLv7xHwIAOiPZWnYIqE0AGPxh58tIesPi5b8LAOicIgSsbQ6y54//OZagFgGgmu9PencCALosiWvLCgErDwCvL/e5HQDA0kLASgNANeMfvf8WAMCxJYSAlQWA12f+/z0AgHeVIWD9aj/bH/2vWICVBIBq1C96/zMU/gDgDMmni5oOWHoAqC75yY36AcD5VCOCL4oQ8EPM0VKvAq6u9/1d78fIYxAAwAWMr6c/Dx/GnPRimS73vrX4A8AU8uTb4o30IOZkaQHgaNwv3O0PANNIkkFcTr6NOVlKB+BE6Q8AmFoymNdkwMI7AFXp77D3wNY/AMzL7H2AxR8BHCa3Lf4AMEdHfYB+zGChAWDw8Z9vFL/JLwMAmJ+qDxB/jRks7AjAyB8ALNr0RwGL2wEok4nFHwAWZ4ajgIVMAVTFv7HWPwAsVJL0o5f8v2z/8cO4oMXsABz0/hYAwDL8ZZpdgLkHgKr458IfAFiWfly6+BvvRewAzNRKBAAuKIkbF70meK4B4GjsT/EPAJbugtcEz3sHwLt/AFiJZGtwZXvrvF89twDg3T8ArFjeO/fle/PbAchzN/4BwCoddQHONREwlwBwtOWQbAUAsFqXe7fO82Xz2QFIkhsBANTBue4FmDkAVLf+eeAPANRFv9gFuPGhL5p9B+CwtxUAQI3k//lDXzF7AMjzvwQAUCPJ1oeOAWYKANX2fyTXAgColw+UAWfbATjoufMfAGop/9NZvzrjEYDZfwCop+oYYPC+X506ANj+B4Cau/z+nfrpdwC0/wGg5t4/DTB9AMgT5/8AUGvJtfdNA8zQATi7XAAArFw/LsfE4/qpAsDg43+8Vv1DAYB6e8+O/XQ7APnhVgAA9ZfEHyd9esojAE/+A4BmSOZ3BBAx/k8BADRBf9J9ANPuAJj/B4CmuNTbOv2pCweA1wVAAKApkhic/tTFdwDGrwYBADRHHu8c3V88ACQ9OwAA0CRJ/s7affEAkOfm/wGgUZLB6c9MsQMweZ4QAKit/ukrgWd8HDAA0BCzBoDk9wEANMvvZg4A4RIgAGia/LIjAADonHw8OPlTAQAAOuhCAWCw8e5dwgBA89gBAIAOEgAAoIMuFADS0TANAKDx7AAAQAcJAADQBUkvPfnTaQLA3wMAaDQ7AADQBclBdvKnUwSA/P8GANAsryI9+dNpHgfsCAAAmiVL0+GMOwB5kgYA0CB5evoz0+wApAEANEjy4vRnpugAXNoLAKA58vydtfviAeDVqzQAgAbJH57+zIUDwOsSgSIgADRF793j++nuAUjeTRIAQD2lT4ZzOAIo5YkeAAA0w6NJn5zyJsCxAAAATTChAFiaKgCkPw8fFh+yAABqbvKx/QzPAsjtAgBA3R3Gw0mfniEAJP8cAECdPTp9BfAb0weAy+NhAAB19t61euoAkI6GabgPAADq62A8/wDw2ncBANRPnqfF9n/6vl+eLQAklxwDAEAdJcnds355pgCQPvmnvfeNFwAAK3Rwdldv1iOAMA0AALXz6Kzt/9LsAeBg/F24FAgA6iOv1uYzXYoZZdnoZX9t4z8WZw2fBgCwWmX57+nw5oe+bA5HAFE+HVAZEADq4APlvzfmEgCOng2gDAgAK3dwvov65rMDUMl3AwBYnTy++1D57425BQC7AACwYofjc78Zn+MOQMkuAACsxAXe/ZfmGgDsAgDACuR5epF3/6U57wAULucfHD0AAOYoSe5d5N1/aeZ7AE7Lfh1l/bWNNfcCAMASVHP/93figua/A1A6zG+H2wEBYAmm698tJAAU2xBZ5GNHAQCwSGXx7+nwu5jCYnYACsVvaKgQCAALMkXx76SFBYDKUSHQUQAAzF2+e9Hi30kLDQDpqPiNOQoAgPmaYev/jblPAZyWPR+NTAUAwJxUW//5zSwbzbTDvtgjgDfKqYAk0gAAZtPLd2bZ+n/7j4klqKYCLo2vhz4AAMxiN30y3Is5WPgRwBtHFwRd/ak4CvgiAICLyfO76dP738ScLC0AlKo+wPpmUrzcCgDgnPK9aW77O8tyOgAnpD9/fzuS/F4AAB9Wlv4O8rku/qWlB4DKq/xWmWYCADhLFof59XmU/k5bSQCoSoGXizRjMgAAzrK7iMW/tJodgHh9SVCMyy0NkwEA8K7d4tj8TizIUkuAp2X7o2f9j67+EpFsBwBwpLrp7/uvY4FWGgBKRQjYMxkAAK8d3fS3k2Wjl7FAKw8ApWz/8cP+Rxu/L3YCrgUAdNXR4l+W/p7Fgq2sA/COcjJAKRCALpvTNb/n+ldFTby9LlgIAKCbvp7XNb/nkUTNDD7evhZ578cAgO7YrS7KW6JadABOqiYD1jdfFC8/CwBov2Gx+H8VS1a7AFDK9h//0F/bWIsk+TQAoK2OSn+fL7rxP0ntjgBOGlzZeVD8FrcCANrmuPGfxgrUZwpgkgPXBQPQUkts/E/810eNvZ0McF0wAO2y1Mb/JPXeAYjfPDMAANpgoXf8n1ctS4CnZfuj1GQAAC3wqFj8b0QNNCIAlEwGANBox3f81+JYu9ZTAJOYDACggbI4GH+yytLfabXvALzDZAAATZOPb9Zp8S81LgCYDACgYXbTp8Nh1EzjjgDeGFzZ3iryy4MAgLrK87vp0/u3ooYaUwI87fVkQBlgtgIAaiffKxb/2o6xNzYAlLL9xw9NBgBQO8d3/Nf2uLqxRwAnDa7s/Fj8p1wLAFi92jX+J2neFMAkl00GAFATNWz8T9KKAHDiumCTAQCsUi0b/5M0ugNwUrY/etb/6OovxVHAdgDAsh01/r+JhmhNACgVIWDPZAAAy1fvxv8k7egAnJD+/P3tSPJ7AQDLUDb+y1tqG6Z1AaDyKr9VprEAgMXK4jC/3oTS32mtDADVdcEmAwBYvN0mLv6ldu4AhMkAABZutzh2vhMN1aoS4GkmAwBYkGGx+H8VDdbqAFAyGQDAXB1f8/syGqz1AaBUPTNgfbO8KngjAGBaR4t/Wfp7Fg3X2g7AOw7GN5UCAZhJL99paunvtM4EgGoy4NL4uhAAwJS+Tp8MWzNi3oqnAV7E4OPta5H3fgwAOL/d6qK5FulEB+CkajJgffNF8fKzAIAPa3zjf5LOBYBStv/4h/7axlokyacBAO/Tksb/JJ07AjhpcGXnQfEt2AoAOO248Z9GC3VnCmCSA9cFA/AeLWr8T9LpAPB2MsB1wQD8Vqsa/5N0ewcgfvPMAAAoNfqO//PqZAnwtGx/lJoMAKDwqFj8b0QHCACvmQwA6Lij0t9Olo06cSzc6SmASUwGAHRSFgfjT9pc+jut8x2Ad5gMAOiefHyzS4t/SQA4xWQAQOfspk+Hw+gYHYAJsl9HWX/t6k+RJF8EAO2V53fTp/e/iQ4SAN4jez4a9dc3y47EVgDQQvlesfh3dgxcADhDtv/4ockAgBY6vuO/s8e9pgDOYXBl58fiW3UtAGiDzjX+J1ECPI/LJgMAWqODjf9JBIBzOHFdsMkAgGbrZON/Eh2Ac8r2R8/6H139pTgK2A4AmqfDjf9JBIALKELAnskAgCbqduN/EkcAF5T+/P3tSPJ7AUAzlI3/8pZXfkMAmMar/FaZJgOAusviML+u9PcuAWAK1XXBJgMAmmDX4j+ZADAlkwEAtbdbHNveCSZSApyByQCA2hoWi/9XwXsJADMyGQBQM8fX/L4M3ksAmIPqmQHrm+VVwRsBwOocLf5l6e9ZcCYdgHk5GN9UCgRYsV6+o/R3PgLAnFSTAZfG14UAgJX5On0yNKJ9Tp4GOGeDj7evRd77MQBYpt3qojbOTQdgzqrJgPXNF8XLzwKAZdD4n4IAsADZ/uMf+msba5EknwYAi6PxPzVHAAs0uLLzoPgWbwUA83fc+E+DC1MCXKQD1wUDLIzG/0wEgAV6OxngumCAedP4n5EAsGAnnhkAwHy4438OlACXINsfpSYDAObiUbH43whmJgAsickAgBkdlf52smzkWHUOTAEsmckAgKlkcTD+ROlvfnQAls1kAMDF5eObFv/5EgCWzGQAwIXtpk+Hw2CudABWIPt1lPXXrv4USfJFAPB+eX43fXr/m2DuBIAVyZ6PRv31zbKDsRUATJDvFYu/MeoFEQBWKNt//NBkAMAEx3f8Oy5dEFMANTC4svNj8UdxLQAoafwvgRJgHVw2GQDwlsb/UggANXDiumBbXUDXafwviQ5ATWT7o2f9j67+UhwFbAdAF2n8L5UAUCNFCNgzGQB0k8b/sjkCqJn05+9vR5LfC4CuKBv/5S2pLJUAUEev8ltlGg6A9sviML+u9Ld8AkANVdcFmwwAumHX4r8aAkBNmQwAOmC3OPa8E6yEEmCNmQwAWmxYLP5fBSsjANScyQCgdY6v+X0ZrIwA0ADVMwPWN8urgjcCoMmOFv+y9PcsWCkdgKY4GN9UCgQar5fvKP3VgwDQENVkwKXxdSEAaLCv0ydDI841IQA0SDUZkBc7AQDNo/FfMzoADZPtj9L++uaL4uVnAdAMGv81JAA0ULb/+If+2sZaJMmnAVBnGv+1lQSNNbiy86D4I9wKgDo6bvynQe3oADTZgeuCgRrT+K81AaDB3k4GuC4YqB+N/5oTABruxDMDAOpC478BlABbwGQAUCOPisX/RlB7AkBLmAwAVu6o9LeTZSPHkg1gCqBlTAYAK5LFwfgTpb/m0AFoG5MBwCrk45sW/2YRAFrGZACwArvp0+EwaBQdgBbKfh1l/bWrP0WSfBEAi5Tnd9On978JGkcAaKns+WjUX98sOx5bAbAQ+V6x+BtDbigBoMWy/ccPTQYAC3F8x7/jxobSAWi7w/x2mdIDYH4yd/w3nwDQclUp8LLJAGCONP5bQQDogBPXBduqA2al8d8SOgAdke2PnvU/uvpLRLIdANPQ+G8VAaBDihCwZzIAmEqep8Xi/3nQGo4AOib9+fvbkeT3AuC8jhr/14NWEQC66FV+y2QAcE4a/y0lAHSQyQDgAnYt/u0kAHSUyQDgHHaLY8M7QSspAXaYyQDgDMNi8f8qaC0BoONMBgDvOL7m92XQWgIAR88MWN+8VrzcCKDbXjf+i3P/Z0Gr6QBw5GB8UykQiF6+o/TXDQIAlWoy4NL4uhAAnfZ1+mRoRLgjBADeqiYD8mInAOgijf+O0QHgN7L9Udpf33xRvPwsgK7Q+O8gAYB3ZPuPf+ivbaxFknwaQLtp/HdWEvAegys7D4q/IlsBtNNx4z8NOkcHgPc7cF0wtJrGf6cJALzX28kA1wVDG+1q/HebAMCZTjwzAGiP3erR4HSaEiAfZDIAWuVRsfjfCDpPAOBcTAZACxyV/naybORYD1MAXIzJAGisLA7Gnyj98YYOABdjMgCaKR/ftPhzkgDAhZgMgEbaTZ8OhwEn6ABwYdmvo6y/dvWnSJIvAqi3PL+bPr3/TcApAgBTyZ6PRv31zbJDshVATeV7xeJvjJeJBACmlu0/fmgyAGrq+I5/x3VMpAPAbA7z2+W7jADqJHPHPx8iADCTqhR42WQA1IrGP+cgADCzE9cF22qE1dP451x0AJiLbH/0rP/R1V8iku0AVkPjnwsQAJibIgTsmQyAFcnztFj8Pw84J0cAzFX1hLEkvxfA8hw1/q8HXIAAwPy9ym+ZDICl0fhnKgIAc2cyAJZq1+LPNAQAFsJkACzFbnHsdidgCkqALIzJAFioYbH4fxUwJQGAhTIZAAtwfM3vy4ApCQAsXPXMgPXNa8XLjQBm87rxX5z7PwuYgQ4Ay3EwvqkUCHPQy3eU/pgHAYClqCYDLo2vCwEwk6/TJ0MjtsyFAMDSVJMBebETAExD45+50gFgqbL9Udpf33xRvPwsgPPS+GfuBACWLtt//EN/bWMtkuTTAM6m8c+CJAErMriy86D4K7gVwGTHjf80YM50AFidA9cFw5k0/lkgAYCVeTsZ4LpgmGRX459FEgBYqRPPDACO7VaP1oYFUgJk5UwGwG88Khb/GwELJgBQCyYDIN6U/naybORYjIUzBUCtmAygw7I4GH+i9Mey6ABQLyYD6Kp8fNPizzIJANSKyQA6ajd9OhwGLJEOALWT/TrK+mtXf4ok+SKg7fL8bvr0/jcBSyYAUEvZ89Gov75ZdlS2Alor3ysWf2OwrIQAQG1l+48fmgygtY7v+HfcxUroAFBvh/nt8l1SQLtk7vhn1QQAaq0qBV42GUDLaPxTAwIAtXfiumBbpbSBxj+1oANAI2T7o2f9j67+EpFsBzSVxj81IgDQGEUI2DMZQGPleVos/p8H1IQjABqlekJakt8LaJKjxv/1gBoRAGieV/ktkwE0iMY/tSQA0DgmA2iYXYs/dSQA0EgmA2iI3eLY6k5ADSkB0lgmA6i5YbH4fxVQUwIAjWYygFo6vub3ZUBNCQA0XvXMgPXNa8XLjYBVe934L879nwXUmA4A7XAwvqkUSC308h2lP5pAAKAVqsmAS+NyzlopkFX6On0yNKJKIwgAtMaJyQBYBY1/GkUHgFbJ9kdpf33zRfHys4Dl0fincQQAWifbf/xDf21jLZLk04BF0/inoZKAlhpc2XlQ/BXfCliU48Z/GtAwOgC014HrglkwjX8aTACgtUwGsGC7Gv80mQBAq5kMYEF2q0dTQ4MpAdJ6JgOYs0fF4n8joOEEADrBZABzcVT628mykWMlGs8UAJ1iMoAZZHEw/kTpj7bQAaBbTAYwrXx80+JPmwgAdIrJAKa0mz4dDgNaRAeAzsl+HWX9tas/RZJ8EfAheX43fXr/m4CWEQDopOz5aNRf3yw7MFsB75XvFYu/MVJayREAnVXNcSf5vYBJysZ/2RmBlhIA6LZX+a3yXV7Ab2Xu+KftBAA6rSoFXjYZwCka/3SAAEDnnbgu2GQAJY1/OkEJEKK6LvhZ/6Orv0Qk20F3afzTIQIAvFaEgD2TAR12dM3vf82y0cuADhAA4IRs//HD/kcbvy92Aq4F3XG0+Jelv2cBHaEDAKeZDOgajX86SQCAU0wGdM6uxZ8uEgBgApMBnbGb/vz9nYAO0gGA9zAZ0HrDYvH/KqCjBAA4g8mAljoq/X2u8U+XCQDwAdVkwPpmORWwETSfxj9UdADgPA7GN5UCW6KX7yj9gQAA51JNBlwaXw+lwKb7On0yNOIJIQDAuZ2YDKCZNP7hBB0AuIBsf5T21zdfFC8/C5pE4x9OEQDggrL9xz/01zbWIkk+DepP4x8mEgBgCtnz0b96ZkADaPzDe+kAwLQ8M6Du3PEPZxAAYErVZMBBft14YC1lkYwt/nAGAQBm8HY8UAiol3x807gfnE0AgBlV44FCQF1k1eL/dDgM4ExJAHMx2NgexGHvQeQxCFbhaNvfO384FzsAMCfVTsCr8SeKgStQtv0t/nAhdgBgAQZ/2LkTSfKXYPGOR/3SAM7NPQCwANU9AR4jvAyPzPnDdOwAwAIN/rC9Hb3e3/QCFiDP76ZP798KYCoCACyYcuDcafrDHAgAsCSDK3++XXz4azCLR3EwvuG8H2YnAMASDa5sb0XS+9ZuwIVl4XG+MFdKgLBE1eOE/+HqvSJ6/3tPEzy38l3/5+n/uf+vAcyNHQBYEd2AD/KuHxZIAIAVG3z85xtRdgMEgTfKhf9u8a7/TvWsBWAhBACoCUGgkMd3cTjeVfKDxRMAoEZeHwtsRdeCgIUflk4AgJrqwI6ArX5YIQEAam5w5b9sRXJ4I/Lky2iHR8WPYbHwf2fhh9URAKAh3h4P5HGj+Omfoln+Xvz47vWinwawcgIANNBxGEi2iwP0Mgz0o37Kd/oPI7k0TJ/8k8f0Qs0IANAC1TFBjK8VL4tAEH+M5QeCciv/3yLPi4W+9zAODx/a3od6EwCghQYf/+O1GI8HkeRFKEiKH3kZCOYRDMpF/UXxz9yLfJxGr/gYl/bi1avUgg/NIgBAhwwG2/343e8GxTv1fvWjlx8FgtOTBkmkb1/nvTQODqqfO78HAIAG+/8iI5lYY6P7UgAAAABJRU5ErkJggg==",
        }
      }
      style={
        Object {
          "height": 9,
          "position": "absolute",
          "resizeMode": "stretch",
          "right": 15,
          "width": 11,
        }
      }
    />
  </View>
</View>
`;
